examples/opensora_hpcai/scripts/inference.py

```diff
@@ -143,39 +144,61 @@ def main(args):
     else:
         captions = args.captions
 
-    # split for data parallel
-    captions, base_data_idx = data_parallel_split(captions, rank_id, device_num)
-    print(f"Num captions for rank {rank_id}: {len(captions)}")
+    captions = process_prompts(captions, args.loop)  # in v1.1 each loop can have a different caption
+    captions, base_data_idx = data_parallel_split(captions, rank_id, device_num)  # split for data parallel
+    if args.use_parallel:
+        print(f"Num captions for rank {rank_id}: {len(captions)}")
 
     # 2. model initiate and weight loading
     # 2.1 latte
-    logger.info("STDiT init")
-
     VAE_T_COMPRESS = 1
     VAE_S_COMPRESS = 8
     VAE_Z_CH = SD_CONFIG["z_channels"]
-
-    if isinstance(args.image_size, list):
-        if len(args.image_size) > 2 or args.image_size[0] != args.image_size[1]:
-            raise ValueError(f"OpenSora-v1 support square images only, but got {args.image_size}")
-        args.image_size = args.image_size[0]
+    img_h, img_w = args.image_size if isinstance(args.image_size, list) else (args.image_size, args.image_size)
 
     input_size = (
         args.num_frames // VAE_T_COMPRESS,
-        args.image_size // VAE_S_COMPRESS,
-        args.image_size // VAE_S_COMPRESS,
+        img_h // VAE_S_COMPRESS,
+        img_w // VAE_S_COMPRESS,
     )
-    if args.image_size == 512 and args.space_scale == 0.5:
-        logger.warning("space_ratio should be 1 for 512x512 resolution")
     model_extra_args = dict(
         input_size=input_size,
         in_channels=VAE_Z_CH,
-        space_scale=args.space_scale,  # 0.5 for 256x256. 1. for 512
-        time_scale=args.time_scale,
+        model_max_length=args.model_max_length,
         patchify_conv3d_replace="conv2d",  # for Ascend
         enable_flashattn=args.enable_flash_attention,
     )
-    latte_model = STDiT_XL_2(**model_extra_args)
+
+    if args.model_version == "v1":
+        model_name = "STDiT"
+        if img_h != img_w:
+            raise ValueError(f"OpenSora v1 support square images only, but got {args.image_size}")
+
+        if args.loop > 1:
```

Comment:
i think this arg check should be placed before `process_prompts`. Otherwise, for v1.0, the "looped" captions can still contain two copies.

Reply:
fixed